Add devfile, CONTRIBUTING.md file, set up image build for dev container. (#68)
eclipse/che-plugin-broker: CONTRIBUTING.md

### Development flow

> You can do che-plugin-broker development inside Eclipse CHE workspace.

Reviewer: s/CHE/Che/g. It's not an acronym, and you shouldn't be SHOUTING. :)

To install Eclipse CHE you need to choose infrastructure(openshift, kubernetes)

> and [set up it](https://www.eclipse.org/che/docs/che-7/che-quick-starts.html#setting-up-a-local-kubernetes-or-openshift-cluster).

Reviewer 2: ```suggestion
and [set it up](https://www.eclipse.org/che/docs/che-7/che-quick-starts.html#setting-up-a-local-kubernetes-or-openshift-cluster).
```

To create development Eclipse CHE workspace we provide che-plugin-broker devfile [devfile.yaml](devfile.yaml).
> See more about [devfile](https://redhat-developer.github.io/devfile)

> ### Create Eclipse CHE workspace from devfile

Reviewer: Again, it's Eclipse Che, not CHE! Shh. No yelling. :)

To start Eclipse CHE workspace, [install the latest chectl](https://www.eclipse.org/che/docs/che-7/che-quick-starts.html#installing-the-chectl-management-tool) and start new workspace from devfile:

```shell
$ chectl workspace:start --devfile=https://raw.githubusercontent.com/eclipse/che-plugin-broker/master/devfile.yaml
```

> Open link to the workspace. After workspace start Eclipse CHE editor

Reviewer 2: ```suggestion
Then, open the link to the workspace. After the workspace starts, the Eclipse Che editor
```

clones che-plugin-broker source code to the folder `/projects/src/github.com/eclipse/che-plugin-broker`.
There are two development linux containers inside workspace: `dev` and `plugin-registry`.

### Dev container target

`dev` container created for development che-plugin-broker. It contains pre-installed development binaries: 

> golang, dep tool, git, golangci-lint and so on. In the devfile mounted  volume `/plugins` to the `dev` container 

Reviewer 2: ```suggestion
golang, dep tool, git, golangci-lint and so on. In the devfile, `/plugins` is mounted as a volume on the `dev` container 
```

to store plugins binaries downloaded with help of `unified` plugin broker.

### Plugin registry container target

> `plugin-registry` container it's micro-service to serve Eclipse CHE plugins meta.yaml definitions. 

Reviewer 2: ```suggestion
`plugin-registry` container is a micro-service to serve Eclipse CHE plugins meta.yaml definitions. 
```

> devfile defines this container in the workspace like plugin-registry service exposed in the internal container's network.

Reviewer 2: ```suggestion
The devfile defines this container in the workspace, and the plugin-registry's service is exposed in the internal container's network.
```

> `unified` plugin broker can connect to this service to get plugins meta.yaml information.

Reviewer 2: ```suggestion
The `unified` plugin broker can connect to this service to get plugins meta.yaml information.
```

### Development commands

> devfile.yaml provides development `tasks` for Eclipse CHE workspace.

Reviewer 2: ```suggestion
devfile.yaml provides development `tasks` for the Eclipse Che workspace.
```

> List development tasks defined in the devfile `commands` section.

Reviewer 2: ```suggestion
These are defined in the devfile `commands` section.
```

> To launch development commands in the Eclipse CHE workspace there are three ways:

Reviewer 2: ```suggestion
To launch development commands in the Eclipse CHE workspace, you can:
```

1. `My Workspace` panel. In this panel you can find development tasks and launch them by click.

2. `Run task...` menu: click `Terminal` menu in the main toolbar => click `Run task...` menu => select task by name and click it.
> Notice: also you can find menu `Run task...` using command palette. Type shortcut `Ctrl/Cmd + Shift + P` to call command palette, type `run task`.

3. Manually type task content in the terminal: `Terminal` => `Open Terminal in specific container` => select container with name `dev` and click Enter.
> Notice: use correct working dir for commands in the terminal.

> ### Compilation plugin brokers

Reviewer 2: ```suggestion
### Compiling plugin brokers
```
